contrib/linux-kernel/lib/zstd/compress.c

```diff
@@ -602,7 +600,22 @@ ZSTD_STATIC size_t ZSTD_compressSequences(ZSTD_CCtx *zc, void *dst, size_t dstCa
 	BYTE *op = ostart;
 	size_t const nbSeq = seqStorePtr->sequences - seqStorePtr->sequencesStart;
 	BYTE *seqHead;
-	BYTE scratchBuffer[1 << MAX(MLFSELog, LLFSELog)];
+
+	U32 *count;
+	S16 *norm;
+	U32 *workspace;
+	size_t workspaceSize = sizeof(zc->tmpCounters);
+	{
+		size_t spaceUsed = 0;
+		count = (U32 *)((BYTE *)zc->tmpCounters + spaceUsed);
+		spaceUsed += sizeof(U32) * (MaxSeq + 1);
+		norm = (S16 *)((BYTE *)zc->tmpCounters + spaceUsed);
+		spaceUsed += sizeof(S16) * (MaxSeq + 1);
+		spaceUsed = ALIGN(spaceUsed, sizeof(U32));
+
+		workspace = (U32 *)((BYTE *)zc->tmpCounters + spaceUsed);
```

Comment:
If a `U32*` ptr is created from a `BYTE*` ptr,
even if the initial source was a `U32*`,
it's still a concern for `-Wcast-align`.

See "doubleCast" example :
https://godbolt.org/g/qrWbLq

Solution : express `spaceUsed` in `U32` increment.